cmd/client.go

```diff
@@ -0,0 +1,63 @@
+package cmd
+
+import (
+	"net/http"
+
+	"github.com/exoscale/egoscale"
+)
+
+// roundTripper implements the http.RoundTripper interface and allows client
+// request customization, such as HTTP headers injection. If provided with a
+// non-nil rt parameter, it will wrap around it when performing requests.
+type roundTripper struct {
+	reqHeaders http.Header
+	rt         http.RoundTripper
+}
+
+func newRoundTripper(rt http.RoundTripper, headers map[string]string) roundTripper {
+	var roundTripper = roundTripper{
+		rt:         http.DefaultTransport,
+		reqHeaders: http.Header{},
+	}
+
+	if rt != nil {
+		roundTripper.rt = rt
+	}
+
+	for k, v := range headers {
+		roundTripper.reqHeaders.Add(k, v)
+	}
+
+	return roundTripper
+}
+
+func (rt roundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
```

Comment:
do you use this func?

Reply:
The Go `net/http` uses it ;) https://golang.org/src/net/http/client.go#L250

Reply:
https://golang.org/src/net/http/client.go#L140 oh yeah you have to implem it my bad!